addon/-private/utils/normalize-record-properties.ts

```diff
@@ -82,17 +84,17 @@ function normalizeRelationship(
     relationship.data = [];
     for (let identity of value) {
       if (typeof identity === 'string') {
-        relationship.data.push({ type, id: identity });
+        relationship.data.push({ type: `${type}`, id: identity });
```

Comment:
This will not work if I pass a list of ids to a polymorphic relationship. It should be an error. We should assert for that case.

Reply:
Ah, thank you, was expecting a comment here. I was a little confused as to this case, I'd not remembered seeing an example of someone passing in only IDs.

Reply:
I agree, passing only ids is maybe not the best practice. Maybe we should deprecate it. @dgeb what do you think?

Reply:
Yeah, I'd be glad to deprecate passing in only IDs. Let's get a deprecation in for v0.17 and remove support in v0.18.

Reply:
Ok, I've added the assert and unit test for that. 
Yeah, I actually don't see a test for accepting strings, so I'm going to guess nobody every uses it - deprecation seems appropriate.
Is that something I should do in this PR, or will one of you do the deprecation separately?

Reply:
Oh, I also have a branch with this fix for 0.16 backport (because that's where my project is at.) But I was not sure what to do for that as there is only a 0.16.6 tag and not a branch.